### PR TITLE
test(j2cl): harden viewport metadata edge cases

### PR DESCRIPTION
**Pull request by vega113**
## Summary
- Adds focused J2CL selected-wave viewport regression coverage for metadata-only fragment deltas, wave switches, mixed metadata+blip replacement, non-blip documents, backward growth, and document merge version bounds.
- Documents the `b+` document-id convention, metadata-only routing, legacy fallback TODO under #904, and why document-only updates do not widen known fragment windows.
- Follow-up to merged PR #992 for issue #967 after the review loop completed after the merge race.

## Verification
- `sbt -batch "testOnly org.waveprotocol.box.server.frontend.ViewportLimitPolicyTest org.waveprotocol.box.server.rpc.FragmentsServletViewportLimitTest org.waveprotocol.box.server.frontend.WaveClientRpcViewportHintsTest" j2clSearchBuild j2clSearchTest`
- `rg "extractContentEntries|extractReadBlips|blipIdFromSegment" -n j2cl/src/main/java j2cl/src/test/java` returned no matches
- `git diff --check`
- `python3 scripts/validate-changelog.py`

Claude review: final focused Opus review returned `conclusion: pass`; remaining notes were non-blocking follow-up suggestions.

## Summary by CodeRabbit

* **Refactor**
  * Clarified viewport/fragment behavior and centralized internal identifiers for conversational documents; documented fallback handling with a temporary TODO note.

* **Tests**
  * Expanded tests for fragment handling, including metadata-only scenarios, viewport merge directions, document-merge regressions, and segment-specific assertions.

### CODE REVIEW

File: j2cl/src/test/java/org/waveprotocol/box/j2cl/search/J2clSelectedWaveProjectorTest.java [hunk 1]
```diff
@@ -432,6 +404,136 @@ public void projectFallsThroughToDocumentsWhenMetadataOnlyFragmentsHaveNoPreviou
...
+            new SidecarSelectedWaveUpdate(
+                2,
+                WAVELET_NAME,
+                true,
+                CHANNEL_ID,
+                44L,
+                "HASH2",
+                Arrays.asList("user@example.com"),
+                Collections.<SidecarSelectedWaveDocument>emptyList(),
+                metadataOnlyFragments(44L, 40L, 44L)),
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/test/java/org/waveprotocol/box/j2cl/search/J2clSelectedWaveProjectorTest.java:431]: Fixed: added  and used it in the wave-switch test.

File: j2cl/src/test/java/org/waveprotocol/box/j2cl/search/J2clSelectedWaveProjectorTest.java [hunk 2]
```diff
@@ -1017,6 +1226,53 @@ private static SidecarSelectedWaveUpdate sampleUpdate() {
...
+        true,
+        CHANNEL_ID,
+        version,
+        historyHash,
+        Arrays.asList("user@example.com"),
+        Collections.<SidecarSelectedWaveDocument>emptyList(),
+        new SidecarSelectedWaveFragments(
+            version,
+            fromVersion,
+            version,
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/test/java/org/waveprotocol/box/j2cl/search/J2clSelectedWaveProjectorTest.java:1246]: Fixed: clamped to  so the helper never produces a negative fromVersion.
